tests/test_upgrade.py

```diff
@@ -102,3 +102,17 @@ def test_upgrade_install_missing(pipx_temp_env, capsys):
     assert not run_pipx_cli(["upgrade", "pycowsay", "--install"])
     captured = capsys.readouterr()
     assert "installed package pycowsay" in captured.out
+
+
+def test_upgrade_multiple(pipx_temp_env, capsys):
+    name = "pylint"
+    pkg_spec = PKG[name]["spec"]
+    initial_version = pkg_spec.split("==")[-1]
+    assert not run_pipx_cli(["install", f"{pkg_spec}"])
```

Comment:
Looks like the f-string isn't necessary here?
```suggestion
    assert not run_pipx_cli(["install", pkg_spec])
```